pkg/knowledge_base2/operational_rule.go

```diff
@@ -90,6 +99,24 @@ const (
 	ClosestSelectionOperator SelectionOperator = ""
 )
 
+func (rule AdditionalRule) Hash() string {
+	// Convert the struct to a byte slice.
+	// Note that the struct must be able to be converted to JSON,
+	// so all fields must be exported (i.e., start with a capital letter).
+	byteSlice, err := json.Marshal(rule)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This is a disguised panic, not sure how I feel about that especially because we use `zap` for logging.